SCClassLibrary/Common/Control/Server.sc

```diff
@@ -351,13 +343,13 @@ Server {
 
 	}
 
+	maxNumClients { ^maxNumClients ?? { options.maxLogins } }
```

Comment:
why not keep the name `maxLogins`? having two names for the same concept is confusing.

Reply:
there is a difference between the maxLogins the server gets from options.maxLogins, 
and the one it gets sent back from the server process, which is not settable. 
there was an earlier discussion with Julian that I cant find now, which ended with changing numClients to maxNumClients - I preferred maxNumClients semantically because it says it is the maximum number of clients that will be able to login; I found maxLogins less clear in that respect. 
The hidden internal variable is the actual number of logins/clients gotten from the server process, so maybe this should best be actualMaxNumClients or actualMaxLogins or so, because that is what it really is. 
the public method returns actualMaxLogins if already known, or if not, the expected maxLogins as set in options. 
what would be the clearest name for that? just maxLogins?

Reply:
OK, thanks for explaining. Maybe you should take out the fallback. The confusing part of it is that the same method is being used to get two different values, and you really have no way of telling whether you got the want you wanted or not. I think it's not so useful behavior in this case, and could actually be very helpful for catching logic errors if it was left as a pure getter.

w/r/t the name I still don't see the value of using two different names here. Maybe the name `maxLogins` is confusing, but I think what's more confusing is explaining "in server options it's called maxLogins and means _this_, and in the server it's called maxNumClients but means the same thing." Either way, you still have to explain the unclear name. I forget, is `maxLogins` entirely new in 3.9? We could just rename all usages and call it a day. I agree that it's unclear. My vote would be for `maxClients`, because "max" + a plural noun already implies it's a numeric value.

Reply:
Thanks for pointing out that it is confusing - it is obscure by trying to be clever. 
FYI, options and options.maxLogins was there from the very beginning.

So here is a resolution proposal:
* options.maxLogins should never touched - it was set by the process response, but that is wrong, options might be used for several servers, so they should only change by user code. 
* server will store the maxLogins info from the running process as maxLoginsFromProcess 
which is the clearest name I could think of.
* when the server is down, maxLoginsFromProcess is correclty nil; BUT: 
allocators may be made anyway, and then they need a fallbacl value, options.maxLogins.
there are about 8 places where this is needed, so we need a method for it to avoid duplication.
* this method is maxClients { maxLoginsFromProcess !? { options.maxLogins } }
in 99+ percent of all cases, this will be the same number anyway. 
users will want to ask the server for its maxClients while it is off, and options.maxLogins is the expected number at this point. I prefer 'maxClients' over maxClientsOrFallback, but if you have a better name, just let me know. 
- will commit as soon as I have a positive answer that this is agreed on, 
to keep commit history cleaner.

Reply:
@brianlheim - are you happy with this proposal? then I would commit it.

Reply:
I think this sounds good! Thanks for your patience.